lib/linked_list.rb

```diff
@@ -19,38 +19,88 @@ def initialize
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
     def add_first(value)
-      raise NotImplementedError
+      @head = Node.new(value, @head)
     end
 
     # method to find if the linked list contains a node with specified value
     # returns true if found, false otherwise
     def search(value)
-      raise NotImplementedError
+      current = @head 
+
+      until current.nil?
+        return true if current.data == value 
+        current = current.next
+      end 
+
+      return false
     end
 
     # method to return the max value in the linked list
     # returns the data value and not the node
     def find_max
-      raise NotImplementedError
+      return nil if @head.nil?
+      max = 0

```

Comment:
What if it's a linked list of negative values, or non-integers?  

```suggestion
      max = @head.data
```